lms/djangoapps/courseware/date_summary.py

```diff
@@ -195,6 +240,39 @@ def description(self):
     def date(self):
         return self.course.end
 
+    def register_alerts(self, request, course):
+        """
+        Registers an alert if the end date is approaching.
+        """
+        now = datetime.datetime.now(UTC())
```

Comment:
For some reason the course end date in the message and in the right sidebar are different, I believe because in this case the course end date was set to 12am and there is a slight issue with timezones.